File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
     int max = 0;
     int left = 0;
     int right = heights.length-1;
     while(left<right){
        int d = Math.min(heights[left],heights[right]);
        int area = (right - left) * d;
        max = Math.max(max,area);
        if(heights[left] < heights[right]) left++;
        else right --;
     }   
     return max;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int min = int.MaxValue;
        int profit = 0;
        foreach(int price in prices){
            min = Math.Min(min,price);
            profit = Math.Max(profit,price-min);
        }
        return profit;
    }
}